pluginLauncher/lib/source/DobbyRdkPluginManager.cpp

```diff
@@ -578,17 +666,23 @@ bool DobbyRdkPluginManager::runPlugins(const IDobbyRdkPlugin::HintFlags &hookPoi
         return false;
     }
 
-    const auto pluginsInConfig = mContainerConfig->rdk_plugins->names_of_plugins;
-    const int rdkPluginCount = mContainerConfig->rdk_plugins->plugins_count;
-
-    std::string pluginName;
+    // Determine the order of launching based on the dependencies.
+    std::vector<std::string> launchOrder;
+    if (hookPoint < IDobbyRdkPlugin::HintFlags::PostHaltFlag)
+    {
+        launchOrder = mDependencySolver.getOrderOfDependency();
```

Comment:
`getOrderOfDependency()` can return an empty vector if there's a circular dependency or other error. There's no check for an empty vector here so `runPlugins()` will still return true in that scenario. `runPlugins()` should probably return false if a cycle is detected

Reply:
Agreed, done.